File: frontend/src/types/paciente.ts
export type Sexo = "MASCULINO" | "FEMININO" | "NAO_INFORMADO";

export type StatusInternacao = "INTERNADO" | "AMBULATORIAL" | "ALTA" | "OBITO";

export interface Paciente {
  id: string;
  nome: string;
  prontuario: string;
  data_nascimento: string | null;
  sexo: Sexo;
  setor: string | null;
  leito: string | null;
  status_internacao: StatusInternacao;
  observacoes: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface PacienteListagem {
  total: number;
  page: number;
  page_size: number;
  items: Paciente[];
}

export interface PacienteFormData {
  nome: string;
  prontuario: string;
  data_nascimento?: string | null;
  sexo: Sexo;
  setor?: string | null;
  leito?: string | null;
  status_internacao: StatusInternacao;
  observacoes?: string | null;
}

File: backend/app/schemas/paciente.py
"""
Schemas de validação de entrada/saída do módulo Pacientes.
"""
import uuid
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.paciente import SexoEnum, StatusInternacaoEnum


class PacienteBase(BaseModel):
    nome: str = Field(..., min_length=2, max_length=200)
    prontuario: str = Field(..., min_length=1, max_length=50)
    data_nascimento: date | None = None
    sexo: SexoEnum = SexoEnum.NAO_INFORMADO
    setor: str | None = Field(default=None, max_length=100)
    leito: str | None = Field(default=None, max_length=20)
    status_internacao: StatusInternacaoEnum = StatusInternacaoEnum.AMBULATORIAL
    observacoes: str | None = Field(default=None, max_length=1000)


class PacienteCreate(PacienteBase):
    pass


class PacienteUpdate(BaseModel):
    """Todos os campos opcionais - permite atualização parcial (PATCH)."""

    nome: str | None = Field(default=None, min_length=2, max_length=200)
    prontuario: str | None = Field(default=None, min_length=1, max_length=50)
    data_nascimento: date | None = None
    sexo: SexoEnum | None = None
    setor: str | None = Field(default=None, max_length=100)
    leito: str | None = Field(default=None, max_length=20)
    status_internacao: StatusInternacaoEnum | None = None
    observacoes: str | None = Field(default=None, max_length=1000)


class PacienteOut(PacienteBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime


class PacienteListOut(BaseModel):
    """Resposta paginada da listagem de pacientes."""

    total: int
    page: int
    page_size: int
    items: list[PacienteOut]
